### PR TITLE
LATTICE 2419 return uuids with table or column objects

### PR DESCRIPTION
**Pull request by pckeyes**
getExternalDatabaseTables() now returns a Map<UUID, OrganizationExternalDatabaseTable> and getExternalDatabaseTablesWithColumns() now returns a Map<Pair<UUID, OrganizationExternalDatabaseTable>, Map<UUID, OrganizationExternalDatabaseColumn>>. This enables using accessCheckForPrincipals instead of isAuthorized for checking permissions.

### CODE REVIEW

File: src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt [hunk 1]
```diff
@@ -124,14 +124,20 @@ class ExternalDatabaseManagementService(
         return organizations.keys
     }
 
-    fun getExternalDatabaseTables(orgId: UUID): Set<OrganizationExternalDatabaseTable> {
-        return organizationExternalDatabaseTables.values(belongsToOrganization(orgId)).toSet()
+    fun getExternalDatabaseTables(orgId: UUID): Map<UUID, OrganizationExternalDatabaseTable> {
+        return organizationExternalDatabaseTables.entrySet(belongsToOrganization(orgId))
```
Comment (katherinebernstein) [inline src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt]: seems like you should be able to just `return organizationExternalDatabaseTables.entrySet(belongsToOrganization(orgId))`

File: src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt [hunk 2]
```diff
@@ -124,14 +124,20 @@ class ExternalDatabaseManagementService(
...
+                    it.key to it.value
+                }.toMap()
     }
 
-    fun getExternalDatabaseTablesWithColumns(orgId: UUID): Map<OrganizationExternalDatabaseTable, Set<OrganizationExternalDatabaseColumn>> {
+    fun getExternalDatabaseTablesWithColumns(orgId: UUID): Map<Pair<UUID, OrganizationExternalDatabaseTable>, Map<UUID, OrganizationExternalDatabaseColumn>> {
         val tables = getExternalDatabaseTables(orgId)
         return tables.map {
-            it to (organizationExternalDatabaseColumns.values(belongsToTable(it.id)).toSet())
+            Pair(it.key, it.value) to (organizationExternalDatabaseColumns.entrySet(belongsToTable(it.key))
```
Comment (katherinebernstein) [inline src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt]: same thing here with skipping the `.map { .. } .toMap()`

File: src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt (hunk 1 above)
Comment (pckeyes) [reply src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt]: that returns a set<map.entry<uuid, table>>, which is fine too. but I will probably still have to do a map{} in the controller.

File: src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt (hunk 2 above)
Comment (pckeyes) [reply src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt]: same thing again with having to move the map{}s to the controller.

**Pull request merged**